lib/src/signed_video_h26x_sign.c

```diff
@@ -631,29 +580,72 @@ signed_video_add_nalu_part_for_signing_with_timestamp(signed_video_t *self,
+  *sei_size = 0;
+  if (self->num_of_completed_seis < 1) {
+    DEBUG_LOG("There are no completed seis.");
+    return SVI_OK;
+  }
+  // Assign the SEI size to the provided pointer.
+  *sei_size = self->sei_data_buffer[self->num_of_completed_seis - 1].completed_sei_size;
+  if (!sei) return SVI_OK;
+  // Copy SEI data to the provided pointer.
+  memcpy(sei, self->sei_data_buffer[self->num_of_completed_seis - 1].payload, *sei_size);
+
+  // Reset the fetched SEI information from the sei buffer.
+  free(self->sei_data_buffer[self->num_of_completed_seis - 1].payload);
+  --(self->num_of_completed_seis);
+  shift_sei_buffer_at_index(self, self->num_of_completed_seis);
+  return SVI_OK;
+}
+
 SignedVideoReturnCode
-signed_video_get_nalu_to_prepend(signed_video_t *self,
-    signed_video_nalu_to_prepend_t *nalu_to_prepend)
+signed_video_get_sei(signed_video_t *self, uint8_t *sei, size_t *sei_size)
 {
-  if (!self || !nalu_to_prepend) return SV_INVALID_PARAMETER;
 
-  if (self->num_nalus_to_prepend < 1) {
-    DEBUG_LOG("No items in |nalus_to_prepend_list|");
-    return SV_NOT_SUPPORTED;
+  if (!self || !sei_size) return SV_INVALID_PARAMETER;
+  *sei_size = 0;
+  if (self->num_of_completed_seis < 1) {
+    DEBUG_LOG("There are no completed seis.");
+    return SV_OK;
   }
+  // Assign the SEI size to the provided pointer.
+  *sei_size = self->sei_data_buffer[0].completed_sei_size;
+  if (!sei) return SV_OK;
+  // Copy the SEI data to the provided pointer.
+  memcpy(sei, self->sei_data_buffer[0].payload, *sei_size);
+
+  // Reset the fetched SEI information from the sei buffer.
+  free(self->sei_data_buffer[0].payload);
+  --(self->num_of_completed_seis);
+  shift_sei_buffer_at_index(self, 0);
+  return SV_OK;
+}
 
-  int list_item = --(self->num_nalus_to_prepend);
-  DEBUG_LOG("Getting list item %d", list_item);
-  if (list_item < 0 || list_item >= MAX_NALUS_TO_PREPEND) {
-    // Frames to prepend list seems out of sync. Flushing list.
-    free_and_reset_nalu_to_prepend_list(self);
-    return SV_UNKNOWN_FAILURE;
+SignedVideoReturnCode
+signed_video_get_nalu_to_prepend(signed_video_t *self,
+    signed_video_nalu_to_prepend_t *nalu_to_prepend)
+{
+  if (!self || !nalu_to_prepend) return SV_INVALID_PARAMETER;
+  // Directly pass the members of nalu_to_prepend as arguments to signed_video_get_lastest_sei().
+  size_t *sei_size = &nalu_to_prepend->nalu_data_size;
+  // Get the size from signed_video_get_lastest_sei() and check if its success.
+  svi_rc status = get_latest_sei(self, NULL, sei_size);
+  if (SVI_OK == status) {
+    if (*sei_size != 0) {
+      nalu_to_prepend->nalu_data = malloc(*sei_size);
+      nalu_to_prepend->prepend_instruction = SIGNED_VIDEO_PREPEND_NALU;
+    } else {
+      nalu_to_prepend->prepend_instruction = SIGNED_VIDEO_PREPEND_NOTHING;
+    }
+    return svi_rc_to_signed_video_rc(
+        get_latest_sei(self, nalu_to_prepend->nalu_data, &nalu_to_prepend->nalu_data_size));
```

Comment:
It is better coding to set status here
`status = get_latest_sei(self, nalu_to_prepend->nalu_data, &nalu_to_prepend->nalu_data_size);`
and then skip the else below, by moving out that return.